app/controllers/concerns/accounts_controller.rb

```diff
@@ -0,0 +1,41 @@
+class AccountsController < ApplicationController
+  def authenticate
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:user_id] = @user.id
+      redirect_to :topics
+    else
+      @errors = ["Hey stupid, enter your information correctly!"]
+      redirect_to :topics
+    end
+  end
+
+  def login
+    @user = User.new
+  end
+
+  def register
+    @user = User.new
+  end
+
+  def new
+    @user = User.new
```

Comment:
You could look at refactoring all of these `User.new` calls into a `before_action`. 

[Read this.](http://guides.rubyonrails.org/action_controller_overview.html#filters)